pkg/apis/core/validation/validation.go

```diff
@@ -6407,7 +6407,12 @@ func ValidateSecret(secret *core.Secret) field.ErrorList {
 func ValidateSecretUpdate(newSecret, oldSecret *core.Secret) field.ErrorList {
 	allErrs := ValidateObjectMetaUpdate(&newSecret.ObjectMeta, &oldSecret.ObjectMeta, field.NewPath("metadata"))
 
-	allErrs = append(allErrs, ValidateImmutableField(newSecret.Type, oldSecret.Type, field.NewPath("type"))...)
+	// TODO: this is a short term fix, we can drop this patch once we
+	// migrate all of the affected secret objects to to intended type,
+	// see https://issues.redhat.com/browse/API-1800
+	if !OpenShiftValidateSecretUpdateIsTypeMutationAllowed(newSecret, oldSecret) {
```

Comment:
I'm wondering how we could e2e validate that the type change is allowed ?

Reply:
I think the easiest way would be to have an integration test. Let me write one.

Reply:
https://github.com/p0lyn0mial/kubernetes/commit/c0a6f63a2b7a675b7d18b2fe83371d027a559cb0